actions/cis-benchmark

```diff
@@ -0,0 +1,120 @@
+#!/usr/local/sbin/charm-env python3
+import os
+import shlex
+import subprocess
+import sys
+import tempfile
+
+from pathlib import Path
+from shutil import which
+
+from charmhelpers.core import hookenv
+from charmhelpers.fetch.giturl import GitUrlFetchHandler
+from charms.layer import snap
+from charms.reactive import is_flag_set
+
+
+BENCH_CMD = '/usr/local/bin/kube-bench'
+BENCH_PKG = 'github.com/aquasecurity/kube-bench'
+
+
+def fail(msg):
+    hookenv.action_fail(msg)
+    sys.exit()
+
+
+def install():
+    '''Install kube-bench and related configuration.
+
+    Install go if needed. Pull, build, and setup snap-based benchmark
+    configuration.
+    '''
+    go_bin = which('go')
+    if not go_bin:
+        go_bin = '/snap/bin/go'
+        snap.install('go', channel='stable', classic=True)
+
+    go_cache = os.getenv('GOCACHE', '/var/snap/go/common/cache')
+    go_path = os.getenv('GOPATH', '/var/snap/go/common')
+    env = os.environ.copy()
+    env['GOCACHE'] = go_cache
+    env['GOPATH'] = go_path
+    Path(go_path).mkdir(parents=True, exist_ok=True)
+
+    go_cmd = ('{bin} get {pkg} '
+              'github.com/golang/dep/cmd/dep'.format(bin=go_bin, pkg=BENCH_PKG))
```

Comment:
Two things:
1. Couldn't we instead download a kube-bench .deb and install that?
2. Regardless, we're going to need a resource for this. Downloading from GH isn't great. Maybe the resource doesn't have to happen right away, but should eventually.

Reply:
+1 on resource support for both the application and the configs.